pkg/proxy/endpointslicecache.go

```diff
@@ -215,15 +215,11 @@ func formatEndpointsList(endpoints []Endpoint) []string {
 
 // endpointSliceCacheKeys returns cache keys used for a given EndpointSlice.
 func endpointSliceCacheKeys(endpointSlice *discovery.EndpointSlice) (types.NamespacedName, string) {
-	if len(endpointSlice.OwnerReferences) == 0 {
-		klog.Errorf("No owner reference set on endpoint slice: %s", endpointSlice.Name)
-		return types.NamespacedName{}, endpointSlice.Name
+	serviceName, ok := endpointSlice.Labels[discovery.LabelServiceName]
+	if !ok {
+		klog.Errorf("No %s label set on endpoint slice: %s", discovery.LabelServiceName, endpointSlice.Name)
 	}
```

Comment:
verify serviceName is non-zero-length?

Reply:
I had been checking this and a few other conditions after the function was called, but you're right that it makes more sense to include it in the function itself, thanks!